src/core/processing/processingalgorithm.cpp

```diff
@@ -0,0 +1,271 @@
+/***************************************************************************
+  processingalgorithm.cpp - ProcessingAlgorithm
+
+ ---------------------
+ begin                : 22.06.2024
+ copyright            : (C) 2024 by Mathieu Pellerin
+ email                : mathieu at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "processingalgorithm.h"
+
+#include <qgsapplication.h>
+#include <qgsprocessingalgorithm.h>
+#include <qgsprocessingprovider.h>
+#include <qgsprocessingregistry.h>
+#include <qgsvectorlayer.h>
+#include <qgsvectorlayerutils.h>
+
+
+ProcessingAlgorithm::ProcessingAlgorithm( QObject *parent )
+  : QObject( parent )
```

Comment:
```suggestion
  : QObject( parent ) = default
```

not sure ... not an expert on this

Reply:
That syntax doesn't work. I think it's fine to have it as is for now, there will soon be stuff in that constructor anyways.